api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -370,23 +372,28 @@ public DbUser createUser(
   @Override
   public DbUser submitDataUseAgreement(
       DbUser dbUser, Integer dataUseAgreementSignedVersion, String initials) {
-    // FIXME: this should not be hardcoded
-    if (dataUseAgreementSignedVersion != CURRENT_DATA_USE_AGREEMENT_VERSION) {
-      throw new BadRequestException("Data Use Agreement Version is not up to date");
-    }
-    final Timestamp timestamp = new Timestamp(clock.instant().toEpochMilli());
-    DbUserDataUseAgreement dataUseAgreement = new DbUserDataUseAgreement();
-    dataUseAgreement.setDataUseAgreementSignedVersion(dataUseAgreementSignedVersion);
-    dataUseAgreement.setUserId(dbUser.getUserId());
-    dataUseAgreement.setUserFamilyName(dbUser.getFamilyName());
-    dataUseAgreement.setUserGivenName(dbUser.getGivenName());
-    dataUseAgreement.setUserInitials(initials);
-    dataUseAgreement.setCompletionTime(timestamp);
-    userDataUseAgreementDao.save(dataUseAgreement);
-    // TODO: Teardown/reconcile duplicated state between the user profile and DUA.
-    dbUser.setDataUseAgreementCompletionTime(timestamp);
-    dbUser.setDataUseAgreementSignedVersion(dataUseAgreementSignedVersion);
-    return userDao.save(dbUser);
+
+    return updateUserWithRetries(
+        (user) -> {
```

Comment:
the user modifier function should only contain code to update the user object.

Otherwise, everything else will be run multiple times with the retries. In this case, the data use agreement will be created several times.